engine/bot-engine.js

```diff
@@ -475,7 +441,9 @@ function ExecutionContext(c, userData, blocks, appContext) {
                 )
             )
         } else if (instr.schedule) {
-            await ec.c.scheduler.schedule(
+            await appContext.scheduler.schedule(
+                appContext.messenger,
```

Comment:
`appContext` doesn't have a messenger, and should not. Use `c.messengerApi.messenger`